src/Host.h

```diff
@@ -121,6 +121,8 @@ class Host : public QObject
     const QString&     getDiscordApplicationID();
     void               setSpellDic(const QString&);
     const QString&     getSpellDic() { QMutexLocker locker(& mLock); return mSpellDic; }
+    void               setUserDictionaryOptions(const bool useDictionary, const bool useShared);
+    void               getUserDictionaryOptions(bool& useDictionary, bool& useShared) { QMutexLocker locker(& mLock); useDictionary = mEnableUserDictionary; useShared = mUseSharedDictionary; }
```

Comment:
I need to get both values in some cases and doing it via non `const` references seemed a valid (and more atomic - only one locking action) way to do it. The getter needs the lock to prevent it being accessed at the same time as the setter is changing it IMVHO.  Annoyingly we have not overhauled the `Host` class to use a `QReadWriteLock` instead to allow multiple *readers* to get details without stepping on each others toes. Sadly some of the recent code added into the Mudlet project has not even bothered to pay attention to `Host::mLock` so I expect more *nasal daemons* to emerge in the future...

N.B. remember that the spell-suggestions code is triggered by context menu (or at least user right mouse) click code so is asynchronous (*I think*) and is the sort of thing that might come to bite us in the future if we do not care to think about thread safety...